analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MultilineBlocksWithoutBrace.cs

```diff
@@ -1,173 +1,421 @@
+    void HaveNoEffect(bool condition)
+    {
+        if (condition)
+            Act.First(); // Secondary
 
-        void TestIfs(bool a, bool b)
-        {
-            if (a)
-            if (b)
-               Console.WriteLine();
+            Act.Second(); // Noncompliant
+    }
+}
 
-            Console.WriteLine();
+class EmptyStatements // Compliant, we ignore cases where at least one of the statements is empty
+{
+    public void First(bool condition)
+    {
+        if (condition) ; Act.Second();
+    }
+    public void Second(bool condition)
+    {
+        if (condition) Act.First(); ;
+    }
+    public void Both(bool condition)
+    {
+        if (condition) ; ;
+    }
+    public void BlockComment(bool condition)
+    {
+        if (condition) /* comment */; Act.Second();
+    }
+}
+
+class Other
+{
+    void FirstSameLineSecondTooFar(bool condition)
+    {
+        if (condition) Act.First(); // Secondary
+                            Act.Second(); // Noncompliant
+    }
+
+    int WithReturnStatement(bool condition)
+    {
+        while (condition)
+            Act.First();
+        //  ^^^^^^^^^^^^ Secondary
+            return Act.Second();
+        //  ^^^^^^^^^^^^^^^^^^^^
+    }
 
-            if (a)
-            while (b)
-               Console.WriteLine();
+    void ZeroIdentation(bool condition)
+    {
+        if (condition)
+Act.First(); // Secondary
+Act.Second(); // Noncompliant
+    }
 
-            Console.WriteLine();
+    void DesendingIndentation(bool condition)
+    {
+                if (condition)
+            Act.First();
+        Act.Second(); // Compliant, although inconvenient.
+    }
 
-            while (a)
-            if (b)
-               Console.WriteLine();
+    void ShiftedIndentation(bool condition)
+    {
+                if (condition)
+        Act.First(); // Secondary
+            Act.Second(); // Noncompliant
+    }
 
-            Console.WriteLine();
+    void AlternativeIndentation(bool condition)
+    {
+        try {
+          if (condition)
+            Act.First(); // This statement is aligned with the '{' of the try on purpose to fix https://github.com/SonarSource/sonar-dotnet/issues/264
         }
+        finally { }
+    }
 
-        void TestWeirdAlignment(bool a)
-        {
-            try {
-              if (a)
-                Console.WriteLine(); // This statement is aligned with the '{' of the try on purpose to fix https://github.com/SonarSource/sonar-dotnet/issues/264
-            } finally { }
-        }
+    void IfElseWithAlternativeSpacing(bool condition)
+    {
+        var n = 0;
+        if (condition)
+            n = 2;
+        else
+        if (condition)
+            n = 2;
+        else
+        if (condition)
+            n = 2;
+        else
+        if (condition)
+            n = 2;
+        n = 4; // Compliant
     }
+
+    public override bool Equals(object obj)
+    {
+        if (obj is null) return false; // probably irrelevant, just pasting as is
```

Comment:
The comment was for you :) We can remove the line if it is not relevant
```suggestion
```

Reply:
The line is relevant. It should not be removed. Sorry I forgot the clear your remarks.